### PR TITLE
docs: add button success react example

### PR DESCRIPTION
**Pull request by ahsinsiddique**
Add a button contrast React example

### CODE REVIEW

File: articles/components/button/index.asciidoc [hunk 1]
```diff
@@ -162,6 +162,14 @@ include::{root}/frontend/demo/component/button/button-success.ts[render,tags=sni
 ----
 include::{root}/src/main/java/com/vaadin/demo/component/button/ButtonSuccess.java[render,tags=snippet,indent=0,group=Java]
 ----
+
+ifdef::react[]
+[source,tsx]
+----
+include::{root}/frontend/demo/component/button/react/button-contrast.tsx[render,tags=snippet,indent=0,group=React]
```
Comment (sissbruecker) [inline articles/components/button/index.asciidoc:169]: This example block is for the success theme variant, and that's also what you implemented in the new React example. As such the file should be called `button-success.tsx`.

File: articles/components/button/index.asciidoc [hunk 2]
```diff
@@ -162,6 +162,14 @@ include::{root}/frontend/demo/component/button/button-success.ts[render,tags=sni
...
+----
+endif::[]
+----
```
Comment (sissbruecker) [inline articles/components/button/index.asciidoc:172]: This is a duplicate block end, the block has already been closed by the line above the `endif`.

File: frontend/demo/component/button/react/button-contrast.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { reactExample } from 'Frontend/demo/react-example'; // hidden-source-line
+import React, { useState } from 'react';
+import { Button } from '@hilla/react-components/Button.js';
+import { HorizontalLayout } from '@hilla/react-components/HorizontalLayout.js';
+
+function Example() {
+  return (
+    <>
+      {/* tag::snippet[] */}
+      <HorizontalLayout theme="spacing" style={{ alignItems: 'baseline' }}>
```
Comment (sissbruecker) [inline frontend/demo/component/button/react/button-contrast.tsx:10]: `align-items` is not used in any other version of this example, and it doesn't seem to be necessary here either.

Comment (sissbruecker) [inline frontend/demo/component/button/react/button-contrast.tsx:10]: The horizontal layout is not shown in the snippets of the other examples, so there is no need to show it here either.

File: frontend/demo/component/button/react/button-contrast.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { reactExample } from 'Frontend/demo/react-example'; // hidden-source-line
+import React, { useState } from 'react';
```
Comment (sissbruecker) [inline frontend/demo/component/button/react/button-contrast.tsx:2]: `useState` is unused

File: frontend/demo/component/button/react/button-contrast.tsx (hunk 3 above)
Comment (ahsinsiddique) [reply frontend/demo/component/button/react/button-contrast.tsx]: removed `alignItems `

File: articles/components/button/index.asciidoc (hunk 2 above)
Comment (ahsinsiddique) [reply articles/components/button/index.asciidoc]: end block removed.

File: frontend/demo/component/button/react/button-contrast.tsx (hunk 4 above)
Comment (ahsinsiddique) [reply frontend/demo/component/button/react/button-contrast.tsx]: Thanks removed

File: frontend/demo/component/button/react/button-contrast.tsx (hunk 3 above)
Comment (ahsinsiddique) [reply frontend/demo/component/button/react/button-contrast.tsx]: yeah, removed from snippet

**Review by sissbruecker — APPROVED** [review]

**Pull request merged**